app.py

```diff
@@ -104,6 +104,47 @@ def experience(index = None):
             "message": "New experience successfully created",
             "id": len(data['experience']) -1
             }), 201
+    if request.method == 'DELETE':
+        if index and str(index).isnumeric():
+                expId = int(index)
+                if 1 <= expId <= len(data['experience']):
+                    data['experience'].pop((expId - 1))
+                    return jsonify({"message":f"Experience with id {expId} has been successfully deleted"}), 200
+                else:
+                    return jsonify({"error": "Invalid experience id"}), 400
+                
+        else:
+            return jsonify({"error": "Invalid experience id"}), 400
+
+    if request.method == 'PUT':
+        if index and str(index).isnumeric():
+            exp_id = int(index)
+            body = request.json
+
+            # Find the experience to edit
+            experience = None
+
+            # Request validation
+            required_fields = ['title', 'company', 'start_date', 'description', 'logo']
+            if body.get('start_date') and not validate_date_string(body.get('start_date')):
+                return jsonify({"error": "Invalid start date. The format should be `June 2023`"}), 400
+            if(body.get('end_date') and not validate_date_string(body.get('end_date'))):
+                return jsonify({"error": "Invalid end date. Format should be like `June 2023`"}), 400
+            if not validate_request(body, required_fields):
+                return jsonify({"error": "Invalid request. Required attributes are missing"}), 400
+            # Request validation End
```

Comment:
Looks good, but do you think we could abstract this part away like Suelen did in #38?